### ISSUE TITLE
Cannot read property 'insertBefore' of null errors in Opera, Chrome, Mozilla, Safari.

### ISSUE THREAD

**Issue by ArkadiuszGorski** [Esri/jsapi-resources#196]
Esri-Loader Team redirects me here.
**Started issue closed without any help answer:**
https://github.com/Esri/esri-loader/issues/154

**Expected behavior**
User should see map always without any limitations.

Project setup - angular, esri-loader v2.6.0, arcgis js ver. 4.13

**Steps to reproduce**
Create a localhost server however you choose with a simple esri-loader implementation.

`   loadModules(["esri/Map",
        "esri/views/MapView",
        "esri/layers/WebTileLayer",
        "esri/Basemap",
        "esri/geometry/support/webMercatorUtils",
        "esri/geometry/Point",
        "esri/layers/GraphicsLayer",
    ], {}).then (([
           Map, MapView, WebTileLayer, Basemap, webMercatorUtils, Point, GraphicsLayer]
    ) => {
           console.log("MapView:" + MapView) `

The rest of the code doesn't really matter because the error happens during the loading/fetching of the resource files.

**Actual behavior**
If a developer or **CLIENT ON PRODUCTION** refreshes the map frequently while making changes and testing, they can encounter an error originating in the dojo.js script used by the loader.
![image](https://user-images.githubusercontent.com/23194646/66997817-77315c00-f0d3-11e9-99dc-b92b13acfd20.png)

**Resources downloaded when everything goes fine:**
![image](https://user-images.githubusercontent.com/23194646/66998086-e909a580-f0d3-11e9-87ac-b86b085db440.png)

**Resources downloaded on error:**
![image](https://user-images.githubusercontent.com/23194646/66998159-08a0ce00-f0d4-11e9-9c5f-6039b2a22495.png)

**Comment by odoe** [comment]
Do you have a repro app this happens with? This looks like the AMD loader can't add the scripts to the page during load for some reason, but I'm unsure why that would happen. This probably happens [here](https://github.com/dojo/dojo/blob/master/dojo.js#L1770).

I tried with angular-cli and cra without issue after multiple refreshes.

**Comment by ArkadiuszGorski** [comment]
Hi there @odoe I am trying to reproduce minimal version with that error. What I investigated for now: 
It happens on most browsers when i use:
`preloadingStrategy: PreloadAllModules` in my app routing and map is in separate module. 
When i disable it and map module is lazy loaded the problem stay only on Safari browser.

**Comment by ArkadiuszGorski** [comment]
@odoe So I investigated more and more, system is complex. It is hard task to create example. I went thru Your suspect with  [this place in dojo.js](https://github.com/dojo/dojo/blob/master/dojo.js#L1770) and what I know, something is wrong with injecting. I use Pusher.js for some communications, after few refreshes I found that it is trying to inject url to pusher script.. the screenshot for evidence. That caused the error described in ticket. My  question is how to workaround that problem.

![Screen with wrong injection](https://user-images.githubusercontent.com/23194646/69245045-6b3f2b00-0ba6-11ea-9214-58ea69b98f34.png)

**Comment by mattheol** [comment]
Workaround is to add empty script in head section <script></script> or any other script (if all of your scripts are in body).
We have encountered similiar problem recently when we migrated our repo to webpack 5.
This piece of code takes first script in document, which has type != dojo:
```
var ya = x.getElementsByTagName("script");
for (d = 0; !Oa; )
   /^dojo/.test((e = ya[d++]) && e.type) || (Oa = e);
```
Then this script is used in injectUrl method to get parentNode and then insert esri scripts before it. In our case sometimes dynamic chunk script (added temporarily to html document head) was selected. But since dynamic chunks are removed from DOM (parent node becomes undefined) code fails and throws error.
